### ISSUE TITLE
unable to test direct line web chat, due to error 500 on botframework system

### ISSUE THREAD

**Issue by davidparker** [Microsoft/BotFramework-WebChat#743]
Hi

Using direct  line webchat, using ngrok on dev side and all is good.   but when trying to set the bots endpoint to ngrok URL  the API for BotManager spits out error 500 with a message of:

CorrelationID: NWRGpC6lySY=

Checked the Json that is posted and all looks good,   raised on Stackoverflow.  but think this is more of an actual bug on the bot framework system

**Comment by danmarshall** [comment]
Hi @davidparker - can you sign out, sign back in, and try again?

**Comment by davidparker** [comment]
Hi yeah finally got it.  Think it hangs on to the sessions so had to do a
cache dump and full reload.  Rather than refresh and tried over again and
it worked.  Needs some info adding to docs that you need to add the bot I'd
and password to local web configure or it all fails on auth

**Comment by danmarshall** [comment]
@davidparker can you provide a link to the doc that needs updating?

**Comment by davidparker** [comment]
Yeah sure.  Think it is this repo actually the readme.

https://github.com/Microsoft/BotFramework-WebChat/blob/master/README.md

It says to simple add css and js link.   And away you go.   And this is
true.   Except.   On running bot locally and using emulater.  You normally
leave bot I'd and pass blank.     However.   When you use the cloud base
direct line.   And use a locally hosted bot.  You need to treat your local
bot.  As though it was live and provide the credentials there too.
Otherwise you get an error in JavaScript console. And an error about oath.
  On the bot side.   I think that as more and more people experiment.
This would be hard for some people to work out why it fails.  And the extra
info here.   Would probably go a long way to elevate that.

Hope that's not been too pedantic.  Just think these little extra
highlighted spots. Help clear up a lot of understanding about how all the
apis work together

**Comment by danmarshall** [comment]
Ok, then maybe this should be in the Emulator's readme?
https://github.com/Microsoft/BotFramework-Emulator

**Comment by davidparker** [comment]
Not really.  Because I was not using emulator this is why I ended up here.
  Because I wanted to test webchat on local bot.    Emulator.  Just lets
you add a URL and that's it.   But direct line lets you test real world by
allowing setting of userid and username.    The emulator is for very basic
testing only.   But is usually where you start of from. 😀   As in.  You
would do a bot that responds,   Run the bot,    Test in bot emulator.  You
would then get real world and want to know who is logged in.  So you look
to pass over info.    You get led down a road of back channel.    Realise
this is over complicated.   Then decide.   Web chat direct line is best
way.    And then you hit the issue of auth

**Comment by compulim** [comment]
@davidparker do you mean we need to show a better error when the web dev did not provide Direct Line secret/token?

**Comment by davidparker** [comment]
Yeah sure better error handling.  Issues was that it was using react or
some kind of ( server is now client ) framework.   That got an error it did
not deal with.  No matter what changes I made.  It never changed the client
state.  So I had to open Dev tools and do a hard reset and finally got past
everything.   I think a reload of the state on error.  And better
messages.  And well trimming the strings.  Would have all resolved the
issue.  As this issue was.  A space from copy and paste.  Hard to see.
Butbroke the whole process for two hours or so.
